donkeycar/parts/web_controller/web.py

```diff
@@ -249,13 +249,14 @@ def open(self):
 
     def on_message(self, message):
         data = json.loads(message)
-
         self.application.angle = data.get('angle', self.application.angle)
         self.application.throttle = data.get('throttle', self.application.throttle)
         if data.get('drive_mode') is not None:
             self.application.mode = data['drive_mode']
             self.application.mode_latch = self.application.mode
-        self.application.recording = data.get('recording', self.application.recording)
+        if data.get('recording') is not None:
```

Comment:
Just a comment for future improvements. Because you are already getting the entry from the dict, you could use it in the next line, w/o getting it again, i.e.
```python
recording = data.get('recording') 
if recording is not None:
    self.application.recording = recording
    self.application.recording_latch = self.application.recording
```

or even
```python
recording = data.get('recording') 
if recording is not None:
    self.application.recording_latch = self.application.recording = recording
```